### DISCUSSION TITLE
Is `max_restarts: 0` a good default?

### DISCUSSION THREAD

**Discussion by shivaraj-bh** [F1bonacc1/process-compose#242] [Q&A] [answered]
`max_restarts: 0` means restart indefinitely (source: https://f1bonacc1.github.io/process-compose/launcher/?h=max_res#auto-restart-on-exit) when `restart: on_failure`. Is this a good default?

**Comment by F1bonacc1** [comment]
What would you propose as an alternative?

**Reply by shivaraj-bh** [reply]
I was thinking of something finite, if it exceeds that limit, we can change the status to `Crashed`, if I am not wrong, this is what `CrashLoopBackOff` in kubernetes is?

**Reply by F1bonacc1** [reply, marked as answer]
In K8s there is no limit and I don't think there is a way configure something finite there. So the defaul 0 acts more like k8s in this regard.
The incremental backoff that exists in K8s is missing today in Process Compose, but no one asked for it so far.

**Reply by shivaraj-bh** [reply]
Gotcha! That makes sense. I don't see any need for the "CrashLoopBackOff" status yet.
